pytorch_lightning/plugins/training_type/parallel.py

```diff
@@ -135,11 +135,6 @@ def block_backward_sync(self):
             yield None
 
     def teardown(self) -> None:
-        # Un-reference the wrapper if any was used.
-        # todo (tchaton): Add support for all plugins.
-        if isinstance(self.model, DistributedDataParallel):
-            self.model = self.lightning_module
-
```

Comment:
@tchaton from https://github.com/PyTorchLightning/pytorch-lightning/pull/8490/ - do you recall why unwrapping this saves memory in the teardown?

doing this with DDP is at odds with https://github.com/PyTorchLightning/pytorch-lightning/issues/8593 

if there are memory leaks intrinsic in the approach of 8593, should we reconsider this? @SeanNaren

Reply:
Hey @ananthsub. It seems the DDP Reducer was holding some GPU memory and had to be deleted to get back to the initial state.